src/wait-for-element-to-be-removed.js

```diff
@@ -0,0 +1,66 @@
+import {getDocument, getSetImmediate, newMutationObserver} from './helpers'
+
+function waitForElementToBeRemoved(
+  callback,
+  {
+    container = getDocument(),
+    timeout = 4500,
+    mutationObserverOptions = {
+      subtree: true,
+      childList: true,
+      attributes: true,
+      characterData: true,
+    },
+  } = {},
+) {
+  return new Promise((resolve, reject) => {
+    if (typeof callback !== 'function') {
+      reject(
+        'waitForElementToBeRemoved requires a callback as the first parameter',
+      )
+    }
+
+    // Check if the element is not present synchronously,
+    // As the name waitForElementToBeRemoved should check `present` --> `removed`
+    ;(function checkElementPresent() {
```

Comment:
Is there a reason this code needs to be in an IIFE?

Reply:
Not much, I was just hoping to separate functionality.

Reply:
Will it have a bad impact?

Reply:
I guess not. Just not a coding style we have anywhere else in the codebase. Feels unnecessary to me.

Reply:
Ok, will get rid of it.